.github/workflows/sts-mock-pull-request.yml

```diff
@@ -0,0 +1,64 @@
+name: Async-credential Pull Request CI
+on:
+  # Triggers the workflow on pull request events for all branches
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
+    paths:
+      - "backend-api/**"
```

Comment:
this needs to be targetted to the sts-mock/** paths :)

Same for the line below (it needs to point to this file)